internal/controller/test_common.go

```diff
@@ -0,0 +1,148 @@
+package controller
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+
+	k8sclientgo "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+
+	numaflowv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	numaflowversioned "github.com/numaproj/numaflow/pkg/client/clientset/versioned"
+	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	defaultNamespace = "default"
+)
```

Comment:
some of this code is somewhat duplicative of suite_test.go (`BeforeSuite()`) but my plan is that that will be removed once we replace the Ginkgo unit testing (for which "unit" is a struct) with unit testing that makes it so the "unit" is a function